public/components/common/forms/model_tag_array_field/tag_field.tsx

```diff
@@ -228,97 +240,127 @@ export const ModelTagField = ({
           >
-            {tagTypeController.field.value === 'string' ? (
-              <EuiComboBox
-                prepend={
-                  selectedTagGroup || !tagKeyController.field.value ? (
-                    'String'
-                  ) : (
-                    <TagTypePopover value={tagTypeController.field.value} onApply={onApplyType} />
-                  )
-                }
-                placeholder="Select or add a value"
-                isInvalid={Boolean(tagValueController.fieldState.error)}
-                singleSelection={{ asPlainText: true }}
-                options={valueOptions}
-                selectedOptions={
-                  tagValueController.field.value ? [{ label: tagValueController.field.value }] : []
-                }
-                onChange={onStringValueChange}
-                onCreateOption={onValueCreate}
-                customOptionText="Add {searchValue} as a value."
-                onBlur={tagValueController.field.onBlur}
-                inputRef={tagValueController.field.ref}
-                isDisabled={!Boolean(tagKeyController.field.value)}
-              />
-            ) : (
-              <EuiFieldNumber
-                prepend={
-                  selectedTagGroup || !tagKeyController.field.value ? (
-                    'Number'
-                  ) : (
-                    <TagTypePopover value={tagTypeController.field.value} onApply={onApplyType} />
-                  )
-                }
-                placeholder="Add a value"
-                value={tagValueController.field.value}
-                isInvalid={Boolean(tagValueController.fieldState.error)}
-                onChange={onNumberValueChange}
-                onBlur={tagValueController.field.onBlur}
-                inputRef={tagValueController.field.ref}
-                disabled={!Boolean(tagKeyController.field.value)}
-              />
-            )}
+            {createValueField()}
           </EuiFormRow>
         </EuiContext>
       </EuiFlexItem>
-      <EuiFlexItem grow={false} style={index === 0 ? { transform: 'translateY(22px)' } : undefined}>
-        <EuiToolTip content={tags?.length && tags.length > 1 ? 'Remove' : 'Clear'}>
-          <EuiButtonIcon
-            display="base"
-            size="m"
-            iconType="cross"
-            aria-label={`Remove tag at row ${index + 1}`}
-            onClick={() => onRemove(index)}
-          />
-        </EuiToolTip>
-      </EuiFlexItem>
+      {!readOnly && (
+        <EuiFlexItem
+          grow={false}
+          style={index === 0 ? { transform: 'translateY(22px)' } : undefined}
+        >
+          <EuiToolTip content={tags?.length && tags.length > 1 ? 'Remove' : 'Clear'}>
+            <EuiButtonIcon
+              display="base"
+              size="m"
+              iconType="cross"
+              aria-label={`Remove tag at row ${index + 1}`}
+              onClick={() => onRemove(index)}
```

Comment:
The delete confirmation modal for default tags missed. Will we added it in the future PR?
<img width="496" alt="image" src="https://github.com/opensearch-project/ml-commons-dashboards/assets/4034161/a8fad509-ea7f-4826-8da0-83a5e0688725">

Reply:
Yes, I will add it later PR